### ISSUE TITLE
NSInvalidArgumentExceptionBSGURLSessionTracingProxy.m:51

### ISSUE THREAD

**Issue by gonzalonunez** [bugsnag/bugsnag-cocoa#1207]
### Describe the bug
Looks like a crash is happening from within the new network breadcrumbs plugin.

### Steps to reproduce
```
CrashReporter Key:  d9416be80cce66b219bcc4c9ccccf351d909271f
Hardware Model:     iPad7,4
Process:            Primer
Identifier:         com.withprimer.Primer
Version:            1.7.1
Role:               Foreground
OS Version:         iOS 14.7.1


NSInvalidArgumentException: -[NSInvocation getArgument:atIndex:]: index (1) out of bounds [-1, 0]

0  CoreFoundation          -[NSInvocation getArgument:atIndex:]
1  CoreFoundation          -[NSInvocation selector]
2  Primer                  -[BSGURLSessionTracingProxy forwardInvocation:] (BSGURLSessionTracingProxy.m:51:32)
3  CoreFoundation          ____forwarding___
4  CoreFoundation          ___forwarding_prep_0___
5  CFNetwork               __CFNetworkHTTPConnectionCacheSetLimit
6  Foundation              ___NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__
7  Foundation              -[NSBlockOperation main]
8  Foundation              ___NSOPERATION_IS_INVOKING_MAIN__
9  Foundation              -[NSOperation start]
10 Foundation              ___NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__
11 Foundation              ___NSOQSchedule_f
12 libdispatch.dylib       __dispatch_block_async_invoke2
13 libdispatch.dylib       __dispatch_client_callout
14 libdispatch.dylib       __dispatch_continuation_pop$VARIANT$mp
15 libdispatch.dylib       __dispatch_async_redirect_invoke
16 libdispatch.dylib       __dispatch_root_queue_drain
17 libdispatch.dylib       __dispatch_worker_thread2
18 libsystem_pthread.dylib __pthread_wqthread
19 libsystem_pthread.dylib _start_wqthread
```
### Environment
* Bugsnag version: `6.12.1`
* CocoaPods version: N/A (SPM)
* Carthage version (if any): N/A (SPM)
* iOS/tvOS/macOS version(s): iOS 14.7.1
* Simulator or physical device: Physical device
* Xcode version: Xcode 13
* Swift version (if applicable): Swift 5.5

### Example code snippet 

```
  private func startBugsnag() {
    let configuration = BugsnagConfiguration(bugsnagAPIKey)
    configuration.add(BugsnagNetworkRequestPlugin())
    configuration.addOnSendError { event -> Bool in
      event.setCustomContextIfNeeded()
      event.trimStacktraceIfPossible()
      return true
    }
    Bugsnag.start(with: configuration)
  }

// MARK: - BugsnagEvent

private extension BugsnagEvent {
  
  func setCustomContextIfNeeded() {
    guard
      let contextualError = originalError as? ContextualError,
      let customContext = contextualError.customContext
    else {
      return
    }
    context = customContext
  }
  
  func trimStacktraceIfPossible() {
    errors.forEach {
      $0.trimStacktraceIfPossible()
    }
  }
}

// MARK: - BugsnagError

private extension BugsnagError {
  
  /// Represents the number of items we need to remove from a `BugsnagError`'s `stacktrace` before reporting
  /// the error.
  ///
  /// When using `LoggingService`, there are `2` extra items in the stacktrace that we need to account for:
  /// 1. The protocol witness for LoggingService to DefaultLoggingService
  /// 2. The call to notifyError(_:)
  private static let stackTraceItemsToRemoveCount = 2
  
  /// Removes the first `x` items from the `stacktrace` for the receiver, where
  /// `x` is `BugsnagError.stackTraceItemsToRemoveCount`.
  func trimStacktraceIfPossible() {
    guard stacktrace.count > BugsnagError.stackTraceItemsToRemoveCount else {
      return
    }
    stacktrace.removeFirst(BugsnagError.stackTraceItemsToRemoveCount)
  }
}
```

**Comment by gonzalonunez** [comment]
@nickdowell This is great news, thank you! I'll let you know if I'm able to reproduce this in the meantime. Looking forward to the next release :)

**Comment by luke-belton** [comment]
Hi @gonzalonunez - just to let you know the [changes](https://github.com/bugsnag/bugsnag-cocoa/pull/1208) mentioned above were released in [`v6.14.1`](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.14.1) of `bugsnag-cocoa` 🎉 

I'm going to close this issue for now but if you have anything further to add or encounter this crash again in the future then please feel free to re-open! Thanks again for reporting it 😃
